src/metabase/search/appdb/core.clj

```diff
@@ -75,19 +75,25 @@
 
 (defmethod search.engine/results :search.engine/appdb
   [{:keys [search-engine search-string] :as search-ctx}]
+  ;; Check whether there is a query-able index.
   (when-not (search.index/active-table)
-    (when config/is-prod?
-      (log/warnf "Triggering a late initialization of the %s search index." search-engine)
-      (try
-        (future
-          (search.engine/init! search-engine {:force-reset? false}))
-        (catch Exception e
-          (log/error e))))
-    ;; Even if the index exists now, return an error so that we track the issue.
-    (throw (ex-info "Search index is not initialized. Use [[init!]] to ensure it exists."
+    ;; Sync, in case we're just out of sync with the database.
+    (when-not (:active (#'search.index/sync-tracking-atoms!))
+      ;; If there's really no index, and we're running in prod - gulp, try to initialize now.
+      (when config/is-prod?
+        (log/warnf "Triggering a late initialization of the %s search index." search-engine)
+        (try
+          (future
+            (search.engine/init! search-engine {:force-reset? false}))
+          (catch Exception e
+            (log/error e)))))
+    ;; Even if the index exists now, return an error so that we don't obscure that there was an issue.
+    (throw (ex-info "Search Index not found."
                     {:search-engine search-engine
                      :db-type       (mdb/db-type)
-                     :index-state   (search.index/search-engine-appdb-index-state)})))
+                     :version       @#'search.index/*index-version-id*
+                     :index-state   (t2/select :model/SearchIndexMetadata :engine :appdb)})))
```

Comment:
Hmm, it would probably be useful to also show what the state was *before* we sync'd the atom or tried to recover.